### PR TITLE
New config module

### PR DESCRIPTION
**Pull request by danjbh**
## what
* Adding new `config` module that allows us to define our Opsgenie in YAML and pass them all in to a single module

## why
* Many Opsgenie resources have cross-dependencies (e.g. team ID) that are difficult (and sometimes impossible) to manage across different modules
* Simplifies configuration by pulling it into YAML defintions instead of having to define separate modules for each resource

### CODE REVIEW

File: modules/config/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+## Config
+
+A Terraform module that configures a multitude of [Opsgenie resources](https://registry.terraform.io/providers/opsgenie/opsgenie/latest/docs). Many resources have cross-resource dependencies, which may be simpler to handle within a single module in certain cases.
```
Comment (osterman) [inline modules/config/README.md:3]:
```suggestion
A Terraform module that configures a multitude of [Opsgenie resources](https://registry.terraform.io/providers/opsgenie/opsgenie/latest/docs). Many resources have cross-resource dependencies, which may be simpler to handle within a single module in certain cases, such as using YAML configurations.
```

File: modules/config/README.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+This module is designed to accept an input configuration map that is generated by parsing resource definitions in YAML. See below for details & examples.
```
Comment (osterman) [inline modules/config/README.md:5]:
```suggestion
This module is designed to accept an input configuration map. One nice way of handling this by passing resource definitions from a YAML configuration file. See below for details & examples.
```

File: test/Makefile [hunk 3]
```diff
@@ -41,3 +41,9 @@ module: deps
 examples/complete: export TESTS ?= installed lint get-modules get-plugins validate
 examples/complete: deps
 	$(call RUN_TESTS, ../$@)
+
+## Run tests against config module
```
Comment (danjbh) [inline test/Makefile:45]: @osterman I added a separate test here just for `config` -- let me know if that's what you had in mind.

Also, the tests are passing, although it doesn't look like `terratest` is really doing all that much when I inspect the action logs, so I figured I should check with you on that one.

**Comment by danjbh** [comment]
Terratest tests are good to go so I'm merging.. woohoo!

File: examples/config/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,24 @@
+output "api_integrations" {
+  value = module.opsgenie_config.api_integrations
```
Comment (3h4x) [inline examples/config/outputs.tf:2]: Not sure if that's the best way to expose outputs for this particular module.
I didn't do it as api keys should be sensitive outputs.

Comment (3h4x) [reply examples/config/outputs.tf:2]: Never mind I see that it has changed in next PR
https://github.com/cloudposse/terraform-opsgenie-incident-management/pull/6/files

**Pull request merged**